### ISSUE TITLE
Various mesh processing algorithm

### ISSUE THREAD

**Issue by cdcseacave** [nmwsharp/geometry-central#35]
As far as I can tell, there are not yet implemented standard mesh processing algorithms, like:

- decimation (quadratic-error based simplification)
- hole filling
- components detection
- self-intersecting faces detection (preferably correction)
- degenerate face removal
- non-manifold correction (though can be easily implemented as far as I understand using a succession of non-manifold edge and vertex fixing)

Do you have plans to add them at some point?

**Comment by nmwsharp** [comment]
Yes, we absolutely have plans to implement such algorithms! Supporting these kinds of algorithms efficiently was much of the motivation for our mesh data structure. I can't offer any concrete timeline, but development should be active through this Fall.

**Comment by cdcseacave** [comment]
These are great news! I am searching since long time for a library implementing well such algorithms and I still didn't find one. Please let me know when you do, and I can test them on thousands of meshes daily if needed.

**Comment by nickponline** [comment]
@nmwsharp Have any of these been implemented?
